docs/src/architecture/accounts.md

```diff
@@ -1,27 +1,44 @@
 
 <p align="center">
-    <img src="../diagrams/protocol/account/Account_Definition.png">
+    <img src="../diagrams/architecture/account/Account_Definition.png">
 </p>
 
 In the above:
 
-* ID is a unique identifier (32 bytes) of an account which does not change throughout its lifetime. This is not the same as address as accounts do not have addresses.
-* Storage is user-defined data which can be stored in an account. The exact mechanism is TBD. For example, this could be a key-value map or an index-based array.
-* Vault is collection of assets stored in an account. In this collection all fungible assets with the same ID are grouped together, and all non-fungible assets are represented as individual entries.
-* Code is a collection of functions which define an external interface for an account.
+* **Account ID** is a unique identifier (32 bytes) of an account which does not change throughout its lifetime. Account ID consists of 1 [field element](https://0xpolygonmiden.github.io/miden-vm/intro/overview.html) (~64 bits). This field element uniquely identifies a single account and also specifies the type of the underlying account. 
+* **Storage** is user-defined data which can be stored in an account. The exact mechanism is TBD. For example, this could be a key-value map or an index-based array.
+* **Vault** is collection of assets stored in an account. In this collection all fungible assets with the same ID are grouped together, and all non-fungible assets are represented as individual entries.
+* **Code** is a collection of functions which define an external interface for an account.
 
 Functions exposed by the account have the following properties:
 
 * Functions are actually roots of Miden program MASTs (i.e., 32-byte hash). Thus, function identifier is a commitment to the code which is executed when a function is invoked.
 * Only account functions have mutable access to an account's storage and vault. Said another way, the only way to modify an account's internal state is through one of account's functions.
 * Account functions can take parameters and can create new notes.
 
-## Accounts with public and private state
+## Types of Accounts
+There are four types of Accounts in Miden. The first two bits of the Account ID specify the type of the account.
+
+### Regular Account with updatable code
+This account type will be used by most users. They can specify and change their code and use this account for a wallet. The Account ID will start with `00`.
+
+### Regular Account with immutable code
+This account type will be used by most regular smart contracts. Once deployed the code should not change and no one should be able to change it. The Account ID will start with `01`.
+
+### Fungible asset faucet with immutable code
+Assets need to be created by Accounts in Miden. The Account ID will start with `10`. 
+
+### Non-fungible asset faucet with immutable code
+Assets need to be created by Accounts in Miden. The Account ID will start with `11`. 
```

Comment:
I'd probably first introduce the concept of faucets and then describe account types - but this can be done in future PRs.